transport-native-kqueue/src/main/java/io/netty/channel/kqueue/BsdSocket.java

```diff
@@ -80,6 +85,70 @@ long sendFile(DefaultFileRegion src, long baseOffset, long offset, long length)
         return ioResult("sendfile", (int) res);
     }
 
+    int connectx(InetSocketAddress source, InetSocketAddress destination, IovArray data, boolean tcpFastOpen)
+            throws IOException {
+        int flags = tcpFastOpen ? connectResumeOnReadWrite() | connectDataIdempotent() : 0;
```

Comment:
IIUC `connectResumeOnReadWrite()` won't change. if so cache these in a `static final int` to avoid repeated JNI calls.

also consider a `private static final TCP_FO_FLAGS = connectResumeOnReadWrite() | connectDataIdempotent()`

Reply:
They will have to move to `Native` and `KQueueStaticallyReferencedJniMethods` to make sure we initialise our classes correctly, but we can do that.